couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
+    return True
```

Comment:
:+1: . This patch removed the `state.line_length` and change the docstring of `reduce` and `rereduce`
```diff
diff --git a/couchdb/server/__init__.py b/couchdb/server/__init__.py
index 40a4328..f72e855 100644
--- a/couchdb/server/__init__.py
+++ b/couchdb/server/__init__.py
@@ -51,7 +51,6 @@ class BaseQueryServer(object):
         self._config = {}
         self._state = {
             'view_lib': None,
-            'line_length': 0,
             'query_config': {},
             'functions': [],
             'functions_src': [],
diff --git a/couchdb/server/views.py b/couchdb/server/views.py
index e885c65..6ce7ac3 100644
--- a/couchdb/server/views.py
+++ b/couchdb/server/views.py
@@ -79,7 +79,8 @@ def reduce(server, reduce_funs, kvs, rereduce=False):
     :raises:
         - :exc:`~couchdb.server.exceptions.Error`
           If any Python exception occurs or reduce output is twice longer
-          as state.line_length and reduce_limit is enabled in state.query_config
+          than input key-value pairs.
+          In the latter case, we consider it as ``reduce_overflow_error``.
     """
     reductions = []
     _append = reductions.append
@@ -132,7 +133,8 @@ def rereduce(server, reduce_funs, values):
     :raises:
         - :exc:`~couchdb.server.exceptions.Error`
           If any Python exception occurs or reduce output is twice longer
-          as state.line_length and reduce_limit is enabled in state.query_config
+          than input key-value pairs.
+          In the latter case, we consider it as ``reduce_overflow_error``.
     """
     log.debug('Rereducing values:\n%s', values)
     return reduce(server, reduce_funs, values, rereduce=True)
```